STM32Cube/Tasks/controller.c

```diff
@@ -41,25 +42,32 @@ void controlTask(void *argument)
 			airbrakesController.error = airbrakesController.target_altitude - apogeeEstimate;
 			airbrakesController.controller_term_P = airbrakesController.error * CONTROLLER_GAIN_P;
 			float dt = (millis_() - last_ms) / 1000.0; //time delay in s
-			airbrakesController.controller_term_I = airbrakesController.controller_term_I + CONTROLLER_GAIN_I * airbrakesController.error * dt; //Add some time measure here
-			airbrakesController.controller_term_D = CONTROLLER_GAIN_D * (airbrakesController.error - airbrakesController.last_error)/ dt; //here too
+			airbrakesController.controller_term_I = airbrakesController.controller_term_I + CONTROLLER_GAIN_I * airbrakesController.error * dt;
+			if(airbrakesController.controller_term_I > CONTROLLER_I_SATMAX) airbrakesController.controller_term_I = CONTROLLER_I_SATMAX;
+			if(airbrakesController.controller_term_I < -CONTROLLER_I_SATMAX) airbrakesController.controller_term_I = -CONTROLLER_I_SATMAX;
+
+			//prevent divide by 0 errors
+			if(dt < 0.0000001) airbrakesController.controller_term_D = 0;
+			else airbrakesController.controller_term_D = CONTROLLER_GAIN_D * (airbrakesController.error - airbrakesController.last_error) / dt;
+
 			last_ms = millis_();
 			airbrakesController.last_error = airbrakesController.error;
 
-			float extension = airbrakesController.controller_term_P + airbrakesController.controller_term_I - airbrakesController.controller_term_D;
+			float output = airbrakesController.controller_term_P + airbrakesController.controller_term_I - airbrakesController.controller_term_D;
+			float extension = 1.0 - output; //invert the controller output
 
 			if(extension > CONTROLLER_MAX_EXTENSION) extension = CONTROLLER_MAX_EXTENSION;
 			if(extension < CONTROLLER_MIN_EXTENSION) extension = CONTROLLER_MIN_EXTENSION;
 
+			printf_("extension: %f\n", extension);
```

Comment:
can we comment out printfs before merging stuff into master its annoying :thehumanity

Reply:
For the future lets aim too have a variable (or your fancy built into IDE defines) that makes each thread shtfu, but I dont think thats a blocking change